packages/flutter/lib/src/material/dropdown.dart

```diff
@@ -1306,16 +1306,9 @@ class _DropdownButtonState<T> extends State<DropdownButton<T>> with WidgetsBindi
 
     // The width of the button and the menu are defined by the widest
     // item and the width of the hint.
-    List<Widget> items;
-    if (_enabled) {
-      items = widget.selectedItemBuilder == null
-        ? List<Widget>.from(widget.items)
-        : widget.selectedItemBuilder(context);
-    } else {
-      items = widget.selectedItemBuilder == null
-        ? <Widget>[]
-        : widget.selectedItemBuilder(context);
-    }
+    final List<Widget> items = widget.selectedItemBuilder == null
```

Comment:
This appears to just be a compact rewrite of the previous, although I'll be pleased to hear that I'm missing something :-). 

What is the flaw that we're correcting in this PR?

Reply:
We should explicit the type of the specified list so that different types of elements can be added again later.